### ISSUE TITLE
Extention broken for the past week or so

### ISSUE THREAD

**Issue by mattymatty** [d8ahazard/sd_dreambooth_extension#792]
Starting Dreambooth training...
Initializing dreambooth training...
Replace CrossAttention.forward to use default
Instance Bucket 0: Resolution (512, 512), Count: 10
Class Bucket 0: Resolution (512, 512), Count: 300
Target Bucket 0: Resolution (512, 512), Count: 0
We need a total of 0 images.
Nothing to generate.
Preparing dataset
Found 300 reg images.
Preparing dataset
Preparing Dataset (With Caching)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 310/310 [00:28<00:00, 11.00it/s]
Train Bucket 1: Resolution (512, 512), Count: 10
Reg Bucket 1: Resolution (512, 512), Count: 300
Total images: 10
Total dataset length (steps): 10
Sched breakpoint is 500
  ***** Running training *****
  Instance Images: 10
  Class Images: 300
  Total Examples: 20
  Num batches each epoch = 10
  Num Epochs = 100
  Batch Size Per Device = 1
  Gradient Accumulation steps = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Text Encoder Epochs: 100
  Total optimization steps = 1000
  Total training steps = 1000
  Resuming from checkpoint: False
  First resume epoch: 0
  First resume step: 0
  Lora: False, Adam: False, Prec: no
  Gradient Checkpointing: True
  EMA: True
  LR: 2e-06)
Steps:   0%|                                                                                                         | 1/1000 [00:04<1:13:45,  4.43s/it, loss=0.0676, loss_avg=0.0676, lr=2e-6, vram_usage=20.1]OOM Detected, reducing batch/grad size to 0/1.
Traceback (most recent call last):
  File "/home/github/sd5latest/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/memory.py", line 116, in decorator
    return function(batch_size, grad_size, prof, *args, **kwargs)
  File "/home/github/sd5latest/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 861, in inner_loop
    accelerator.backward(loss)
  File "/home/anaconda3/lib/python3.9/site-packages/accelerate/accelerator.py", line 1316, in backward
    loss.backward(**kwargs)
  File "/home/anaconda3/lib/python3.9/site-packages/torch/_tensor.py", line 396, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/anaconda3/lib/python3.9/site-packages/torch/autograd/__init__.py", line 173, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
  File "/home/anaconda3/lib/python3.9/site-packages/torch/autograd/function.py", line 253, in apply
    return user_fn(self, *args)
  File "/home/anaconda3/lib/python3.9/site-packages/torch/utils/checkpoint.py", line 146, in backward
    torch.autograd.backward(outputs_with_grad, args_with_grad)
  File "/home/anaconda3/lib/python3.9/site-packages/torch/autograd/__init__.py", line 173, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: CUDA out of memory. Tried to allocate 1024.00 MiB (GPU 0; 23.63 GiB total capacity; 20.58 GiB already allocated; 339.38 MiB free; 21.10 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Steps:   0%|                                                                                                         | 1/1000 [00:05<1:32:01,  5.53s/it, loss=0.0676, loss_avg=0.0676, lr=2e-6, vram_usage=20.1]
Traceback (most recent call last):
  File "/home/github/sd5latest/stable-diffusion-webui/extensions/sd_dreambooth_extension/scripts/dreambooth.py", line 561, in start_training
    result = main(config, use_txt2img=use_txt2img)
  File "/home/github/sd5latest/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 973, in main
    return inner_loop()
  File "/home/github/sd5latest/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/memory.py", line 114, in decorator
    raise RuntimeError("No executable batch size found, reached zero.")
RuntimeError: No executable batch size found, reached zero.
Training completed, reloading SD Model.
Restored system models.
Returning result: Exception training model: 'No executable batch size found, reached zero.'.

**Comment by ExponentialML** [comment]
> enable either 8bit adam or xformers

While this is true for most cases, there's a memory leak somewhere. I have a 3090 and still get OOM occasionally, even with LoRA, optimizations, etc. I've been looking but can't find where the problem is, but it may be with the way the profiler works.

**Comment by Zuxier** [comment]
> 

4090 here, never ever had a memory leak, i remember once Leppie finding one in early december.
I always train from fresh webui when possible, and the only time i got mid training OOM i was opening youtube videos.
I have been pushing the setting to the max always, so if there was a memory leak i think i would have noticed by now.

**Comment by dcferreira** [comment]
I'm getting a similar error message on a 3080ti, and I've selected all the settings to reduce VRAM usage.

The error message claims it can't find 128MiB (!!!) on a 12GiB GPU. 
I also have closed most processes that use the GPU; task manager reports less than 0.5GiB being used before starting the webui.
Here is the relevant error snippet
```text
Train Bucket 1: Resolution (512, 512), Count: 30
Reg Bucket 1: Resolution (512, 512), Count: 4560
Total images: 30
Total dataset length (steps): 30
Sched breakpoint is 30
  ***** Running training *****
  Instance Images: 30
  Class Images: 4560
  Total Examples: 60
  Num batches each epoch = 30
  Num Epochs = 100
  Batch Size Per Device = 1
  Gradient Accumulation steps = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Text Encoder Epochs: 65
  Total optimization steps = 3000
  Total training steps = 3000
  Resuming from checkpoint: False
  First resume epoch: 0
  First resume step: 0
  Lora: False, Adam: True, Prec: fp16
  Gradient Checkpointing: True
  EMA: False
  LR: 1e-06)
Steps:   0%| | 1/3000 [00:08<6:44:13,  8.09s/it, loss=0.0584, loss_avg=0.0584, lr=1e-6, vram_usage=9.5]OOM Detected, reducing batch/grad size to 0/1.
Traceback (most recent call last):
  File "D:\git_projs\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\memory.py", line 116, in decorator
    return function(batch_size, grad_size, prof, *args, **kwargs)
  File "D:\git_projs\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 807, in inner_loop
    latents = vae.encode(batch["pixel_values"].to(dtype=weight_dtype)).latent_dist.sample()
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\diffusers\models\vae.py", line 566, in encode
    h = self.encoder(x)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\diffusers\models\vae.py", line 134, in forward
    sample = down_block(sample)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\diffusers\models\unet_2d_blocks.py", line 755, in forward
    hidden_states = resnet(hidden_states, temb=None)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\diffusers\models\resnet.py", line 450, in forward
    hidden_states = self.norm1(hidden_states)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\normalization.py", line 272, in forward
    return F.group_norm(
  File "D:\git_projs\stable-diffusion-webui\venv\lib\site-packages\torch\nn\functional.py", line 2516, in group_norm
    return torch.group_norm(input, num_groups, weight, bias, eps, torch.backends.cudnn.enabled)
RuntimeError: CUDA out of memory. Tried to allocate 128.00 MiB (GPU 0; 12.00 GiB total capacity; 9.53 GiB already allocated; 0 bytes free; 9.68 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Steps:   0%| | 1/3000 [00:09<7:47:12,  9.35s/it, loss=0.0584, loss_avg=0.0584, lr=1e-6, vram_usage=9.5]
Traceback (most recent call last):
  File "D:\git_projs\stable-diffusion-webui\extensions\sd_dreambooth_extension\scripts\dreambooth.py", line 561, in start_training
    result = main(config, use_txt2img=use_txt2img)
  File "D:\git_projs\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 973, in main
    return inner_loop()
  File "D:\git_projs\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\memory.py", line 114, in decorator
    raise RuntimeError("No executable batch size found, reached zero.")
RuntimeError: No executable batch size found, reached zero.
Training completed, reloading SD Model.
Restored system models.
Returning result: Exception training model: 'No executable batch size found, reached zero.'.
```

Unlike OP, I didn't have it working before--this is the first time I'm trying dreambooth.

Could I be doing something obviously wrong?

**Comment by Zuxier** [comment]
OP has 24gb of VRAM, you got 12GB, apples to oranges.
I suggest you if possible to join the discord (there is a link in the wiki page), and to follow the wiki eli5 suggestions.
Your settings seems mostly alright, and i think you are more than likely able to train without class images.
Many stuff got changed in the alpha branch and it's def lighter, at the same time is not as stable as main.

**Comment by dcferreira** [comment]
@Zuxier Sorry for spamming this topic again, but I tried joining the discord and the link on the wiki says it expired. Any other link?

**Comment by Zuxier** [comment]
@dcferreira updated the link with one without expiration

**Comment by ExponentialML** [comment]
> > While this is true for most cases, there's a memory leak somewhere. I have a 3090 and still get OOM occasionally, even with LoRA, optimizations, etc. I've been looking but can't find where the problem is, but it may be with the way the profiler works.
> 
> 4090 here, never ever had a memory leak, i remember once Leppie finding one in early december. I always train from fresh webui when possible, and the only time i got mid training OOM i was opening youtube videos. I have been pushing the settings to the max always, so if there was a memory leak i think i would have noticed by now.

Interesting. I think there are just a lot of variables here. Now without gradient accumulation I'm getting good VRAM usage (around 12GB with flash attention) like before. 
I did update my webui install and delete the `__pycache__` directory, but I doubt either or made a significant diference unless there was a bug / caching issue somewhere.

Torch versions, architectures, webui updates (not this extension), and the like make these things very hard to pinpoint, especially with the rapid release cycles.

I'll continue searching, reporting, and fixing anything I find.

**Comment by Hellisotherpeople** [comment]
Really sad to know that the top extension for dreambooth isn't working right now. It appears that no one here even knows why.

**Comment by cschwarzhw** [comment]
> No executable batch size found, reached zero

I'm on 16Gb, and was able to train, using any refernce model, until last week. Sometimes (after a fresh PC boot) I could manage a batch size of 2 or even 4, and a batch of 1 always worked 100% - Now I put in one as a starting point and it always ends up with 'No executable batch size found, reached zero'

**Comment by cschwarzhw** [comment]
> > Makes zero difference in training (it's a default I have always selected but recently I get constant errors regardless). It's not even not applicable anyway as many of the new errors (loading state_dict , Pipeline or config is not set, etc etc) occur when you just try create a blank model, before even starting to try train it. Bottomline is someone fucked with Dreambooth in first week of Jan and now the shit is completely unusable. Kicking myself for leaving git pull enabled in bat.
> 
> Let's not mix up the stuffs, one thing is OOM, another are the issues you are writing about. Work is being done on the ligherlora branch and when its complete it will pushed into main, lot's of improvement in the way. You can always `git checkout` an older commit.

I've seen some other mention git checkout, but profess I'm not very knowledgable there. Would hugely appreciate if you could advise the syntax on running that and nominate a commit from around Jan 10.

**Comment by mykeehu** [comment]
A major revision of the LigherLora branch is in the works, and we look forward to it. It is likely that the current bugs will be fixed after that, but it is also possible that they have already been fixed in this revision. We are waiting.

**Comment by tpcdaz** [comment]
> A major revision of the LigherLora branch is in the works, and we look forward to it. It is likely that the current bugs will be fixed after that, but it is also possible that they have already been fixed in this revision. We are waiting.

Why does google find zero results for this LigherLora or even LighterLora branch?

**Comment by mykeehu** [comment]
Don't look in Google, look in the branches. However, this is still under active development, so things could change at any minute, so consider it a beta version
![image](https://user-images.githubusercontent.com/6390413/213366179-caa42dfa-358b-49ce-a07d-28e948dd56f2.png)

**Comment by tpcdaz** [comment]
> Don't look in Google, look in the branches. However, this is still under active development, so things could change at any minute, so consider it a beta version ![image](https://user-images.githubusercontent.com/6390413/213366179-caa42dfa-358b-49ce-a07d-28e948dd56f2.png)

Ahh thank you :)

**Comment by GunnarQuist** [comment]
I can confirm that something is wrong. I'm sorry I can't provide technical help. All I can say is that recently I've repeatedly failed to train a good checkpoint. When I trained a style it either did not work at all or the generated results were extremely inconsistent. When I reverted Auto1111 and the extension to earlier versions, the trainings worked.

As some of you stated, something amiss upstream might be the likely culprit. I've been satisfied with the work that has been done on this software up until recently. I hope it can be fixed soon.

**Comment by cschwarzhw** [comment]
I saw a [Reddit post ](https://www.reddit.com/r/StableDiffusion/comments/10fej17/automatic1111_dreambooth_broken_with_the_latest/?utm_source=share&utm_medium=ios_app&utm_name=iossmf) with following fix suggestion:
Jan 13 Rollback for Automatic: git checkout a407c9f0147c779865c940cbf62c7019dbc1f7b4
Jan 01 Rollback for Dreambooth: git checkout 4ca69a904f5ddd5651d87032b3dca515eea505ba
Update requirements.txt with gradio==3.16.2

I tried it but Dreambooth is still screw and wont create a model with following:
Exception setting up output: Error(s) in loading state_dict for UNet2DConditionModel:
        Missing key(s) in state_dict: "up_blocks.0.upsamplers.0.conv.weight", "up_blocks.0.upsamplers.0.conv.bias", "up_blocks.1.upsamplers.0.conv.weight", "up_blocks.1.upsamplers.0.conv.bias", "up_blocks.2.upsamplers.0.conv.weight", "up_blocks.2.upsamplers.0.conv.bias".

The Reddit post also advised about a commit named '"@d8ahazard So fu*king sick of thi *hit"  - I havent tried that as I dont know if it's a joke or not.

**Comment by mykeehu** [comment]
Let's be patient a little longer! The programmers are at work, and as I read their developments, they are thoroughly reworking the training to make it better. I'm looking forward to it too, but they've done quite a bit of work on the LigherLora branch. 

Until then, you can revert to an [earlier build as suggested here](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/579#discussioncomment-4558487). I opened a separate folder instead and put only this previous webUI and DB there and no problems with the training on that. Although it doesn't have many convenience features, it can train :)
When the new version comes out, I'll try it again, but I have a working version.

**Comment by GunnarQuist** [comment]
> Let's be patient a little longer!

Hear, hear!

**Comment by Hellisotherpeople** [comment]
@mykeehu I fully appreciate the difficulty of this kind of work and we are extremely happy to see that active development is happening.

However, I am concerned by the possibility that at this point no one may still be quite sure what broke in the first place. Can anyone pipe in with clear reasoning for what happened here? 

Beyond that, I think going forward that we should have some kind of testing done which guarantees a certain "quality" of produced images given certain settings. Hopefully in the future something like that would prevent this situation of random commits breaking this functionality

**Comment by GunnarQuist** [comment]
> I think going forward that we should have some kind of testing done which guarantees a certain "quality" of produced images given certain settings

I have been thinking along those lines. Some sort of standard test training using a half dozen photos of someone that has almost no presence in the latent space. The six dataset images could be made available to everyone along with standard settings that will always produce recognizable and flexible results.

Even better, make that three different test trainings. One of a person, one of a thing, and one of a style. In fact, these test trainings could also serve as infallible tutorials. The perfect answer to the newbie who asks, "how do you do this?"

**Comment by Zuxier** [comment]
@Hellisotherpeople @GunnarQuist 
There are different issues that happened on main, it's hard to pinpoint one, if you were training with [filewords] only you probably had the least issues.
Off the top of my head:
-error related to how the class images were processed
-issues with oom on lora
-class images generation
-gradio error when auto decided to yolo update it
Every issue that gets reported here is usually in the discord hours in advance.
I'm not gonna stay here to write those stuff down while they are discussed in the discord ad nauseam.
Consider as well that most of the work is done by one person, that is doing it out of passion in his free time.
Personally I'm happy that d8 is caring a lot of this project.
Main issues happened while the new branch was getting almost ready to be pushed after it was in the works for quite a while, some early changes were pushed to main, but after some extra reworking, everything just needed more love and it got delayed.
The updates are not forced so i would advise if people are happy with their situation, not to blindly update.
The idea of a standard test training sound good on paper, but when you considers that the most active people in the extension are never on the main branch, and that checking all common combination will require many trainings and not just one.
If either of of you is up for that, feel free to keep track.
Myself i only check for basic stuffs and only on the commit I'm in, and it's usually the commit i suggest in the discord.
At some point in the past (late December)  the extension trained at ludicrous batch sizes for a good 2 weeks, something was off but only few of the people noticed, for the most, the extension had no error and they were happy.
There is a lot more under the hood and not everything is simple to grasp.
On the positive, LigherLora is getting really close to be pushed to main, feel free to join the discord and help us stress test it.

**Comment by mykeehu** [comment]
If you want to get involved in the development, [here is d8's Discord channel](https://discord.com/channels/1049560328654037023/1050814679057969342). The more eyes that see it, the better the code can be.

**Comment by Hellisotherpeople** [comment]
@Zuxier  it sounds like folks do in fact know what's going on! I'm very happy to see that. 

I will join the Discord soon enough!

**Comment by PauloRodrigues33** [comment]
I'm here to say the master branch worked to me with 8GB VRAM and the lastest commit of Stable Difusion WebUI from AUTO1111, (he did a fix for lora, commit: f53527f7786575fe60da0223bd63ea3f0a06a754)
- Lora ON
- Adam opt ON
- xformers ON
- fp16
- only one batch
